src/Sulu/Bundle/MediaBundle/Resources/public/js/components/collections/root/main.js

```diff
@@ -97,22 +104,33 @@ define(['sulumedia/model/media'], function(Media) {
         },
 
         bindCustomEvents: function() {
+            /**
+             * Change current view of the datagrid to given viewKey
+             * viewKey must be specified in listViews-Object
+             */
+            var changeDatagridView = function(viewKey) {
+                this.sandbox.emit('husky.datagrid.view.change', listViews[viewKey].name, listViews[viewKey]['viewOptions']);
+                this.sandbox.sulu.saveUserSetting(constants.listViewStorageKey, viewKey);
+            }.bind(this);
+
             // change datagrid to table
-            this.sandbox.on('sulu.list-toolbar.change.table', function() {
-                this.sandbox.emit('husky.datagrid.view.change', 'table');
-                this.sandbox.sulu.saveUserSetting(constants.listViewStorageKey, 'table');
+            this.sandbox.on('sulu.toolbar.change.table', function() {
+                changeDatagridView('table');
             }.bind(this));
 
             // change datagrid to thumbnail small
-            this.sandbox.on('sulu.list-toolbar.change.thumbnail-small', function() {
-                this.sandbox.emit('husky.datagrid.view.change', 'thumbnail', listViews['thumbnailSmall']['thViewOptions']);
-                this.sandbox.sulu.saveUserSetting(constants.listViewStorageKey, 'thumbnailSmall');
+            this.sandbox.on('sulu.toolbar.change.thumbnail-small', function() {
+                changeDatagridView('thumbnailSmall');
             }.bind(this));
 
             // change datagrid to thumbnail large
-            this.sandbox.on('sulu.list-toolbar.change.thumbnail-large', function() {
-                this.sandbox.emit('husky.datagrid.view.change', 'thumbnail', listViews['thumbnailLarge']['thViewOptions']);
-                this.sandbox.sulu.saveUserSetting(constants.listViewStorageKey, 'thumbnailLarge');
+            this.sandbox.on('sulu.toolbar.change.thumbnail-large', function() {
```

Comment:
@nnatter this could be shortened with ``this.sandbox.on('...', changeDataGridView.bind(this, 'thumbnailLarge'));